dask_ml/feature_extraction/text.py

```diff
@@ -0,0 +1,146 @@
+        if ``analyzer == 'word'``. The default regexp selects tokens of 2
+        or more alphanumeric characters (punctuation is completely ignored
+        and always treated as a token separator).
+    n_features : integer, default=(2 ** 20)
+        The number of features (columns) in the output matrices. Small numbers
+        of features are likely to cause hash collisions, but large numbers
+        will cause larger coefficient dimensions in linear learners.
+    norm : 'l1', 'l2' or None, optional
+        Norm used to normalize term vectors. None for no normalization.
+    binary : boolean, default=False.
+        If True, all non zero counts are set to 1. This is useful for discrete
+        probabilistic models that model binary events rather than integer
+        counts.
+    dtype : type, optional
+        Type of the matrix returned by fit_transform() or transform().
+    alternate_sign : boolean, optional, default True
+        When True, an alternating sign is added to the features as to
+        approximately conserve the inner product in the hashed space even for
+        small n_features. This approach is similar to sparse random projection.
+    """
+    def __init__(self, input='content', encoding='utf-8',
+                 decode_error='strict', strip_accents=None,
+                 lowercase=True, preprocessor=None, tokenizer=None,
+                 stop_words=None, token_pattern=r"(?u)\b\w\w+\b",
+                 ngram_range=(1, 1), analyzer='word', n_features=(2 ** 20),
+                 binary=False, norm='l2', alternate_sign=True,
+                 dtype=np.float64):
+        self.input = input
+        self.encoding = encoding
+        self.decode_error = decode_error
+        self.strip_accents = strip_accents
+        self.preprocessor = preprocessor
+        self.tokenizer = tokenizer
+        self.analyzer = analyzer
+        self.lowercase = lowercase
+        self.token_pattern = token_pattern
+        self.stop_words = stop_words
+        self.n_features = n_features
+        self.ngram_range = ngram_range
+        self.binary = binary
+        self.norm = norm
+        self.alternate_sign = alternate_sign
+        self.dtype = dtype
+
+    def partial_fit(self, X, y=None):
+        """Does nothing: this transformer is stateless.
+        This method is just there to mark the fact that this transformer
+        can work in a streaming setup.
+        """
+        return self
+
+    def fit(self, X, y=None):
+        """Does nothing: this transformer is stateless."""
+        # triggers a parameter validation
+
+        self._get_hasher().fit(None, y=None)
+        return self
+
+    def transform(self, X):
+        """Transform a sequence of documents to a document-term matrix.
+        Parameters
+        ----------
+        X : dask.Bag of raw text documents, length = n_samples
+            Samples. Each sample must be a text document (either bytes or
```

Comment:
It seems like we only support `input='content'`, correct?

I suppose if `input='filename', then `X = db.read_text(X)`, should work.